xayn-ai/src/coi/config.rs

```diff
@@ -7,11 +9,18 @@ pub(crate) struct Configuration {
     pub threshold: f32,
     /// The positive number of neighbors for the k-nearest-neighbors distance.
     pub neighbors: NonZeroUsize,
-    /// The maximum number of key phrases picked during the coi key phrase selection. A coi may have
-    /// more key phrases than this, eg because of merging.
-    pub max_key_phrases: usize,
+    /// The time since the last view after which a coi becomes irrelevant.
+    pub horizon: Duration,
     /// The weighting between coi and pairwise candidate similarites in the key phrase selection.
     pub gamma: f32,
+    /// The penalty for less relevant key phrases in the top key phrase selection.
+    pub penalty: [f32; Self::MAX_KEY_PHRASES],
```

Comment:
We may want to check that the passed configuration for penalty contains finite numbers.

Reply:
we don't have any checks for any value of the configuration yet. i agree that it makes sense to check values before accepting them in the config, i think it makes sense to do this in a separate pr, wdyt?